nav2_bt_navigator/src/navigators/navigate_through_poses.cpp

```diff
@@ -262,6 +275,16 @@ NavigateThroughPosesNavigator::initializeGoalPoses(ActionT::Goal::ConstSharedPtr
   blackboard->set<nav_msgs::msg::Goals>(goals_blackboard_id_,
       std::move(goals_array));
 
+  // Reset the waypoint states vector in the blackboard
+  std::vector<nav2_msgs::msg::WaypointStatus> waypoint_statuses(goals_array.goals.size());
+  for (size_t waypoint_index = 0 ; waypoint_index < goals_array.goals.size() ; ++waypoint_index) {
+    waypoint_statuses[waypoint_index].index = waypoint_index;
```

Comment:
Question: do we need the index if we store the pose and we know they should be ordered, even if remove passed goals edits the 'goals' variable?I think the index is implicit now by the waypoint status vector dimensions

Reply:
Oh I see, this is needed for the WPF, I think we should keep